api/envoy/extensions/compression/common/compressor/v3/compressor.proto

```diff
@@ -0,0 +1,22 @@
+syntax = "proto3";
+
+package envoy.extensions.compression.common.compressor.v3;
+
+import "google/protobuf/any.proto";
+
+import "udpa/annotations/migrate.proto";
+import "udpa/annotations/status.proto";
+import "udpa/annotations/versioning.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.compression.common.compressor.v3";
+option java_outer_classname = "CompressorProto";
+option java_multiple_files = true;
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// [#protodoc-title: Compressor Library]
+
+message CompressorLibrary {
+  // Compressor library specific configuration. See the supported libraries for further
+  // documentation.
+  google.protobuf.Any typed_config = 1;
```

Comment:
Can we move this one to the new `TypedConfig` that will appear in https://github.com/envoyproxy/envoy/pull/10908?

Reply:
Sure, I've subscribed to that PR.

Reply:
I presume it's about `TypedExtensionConfig`, not `TypedConfig`?

Reply:
Ack

Reply:
@htuch If we are moving all the `typed_config`s across the project to the new `TypedExtenstionConfig` message that is being created in #10908, this could be updated then. I'd like to merge this and come back and update when the discussion in #10908 settles, and gets merged. We have been working on this PR for a while, and have a chain of a few other PRs that depend on this one.

Reply:
I don't think we can move them all without a deprecation and major version change. The idea is to make all new ones `TypedExtensionConfig`. Would it satisfy your concern if I submit a quick PR to introduce this independent of #10908?

Reply:
https://github.com/envoyproxy/envoy/pull/11105

Reply:
> I don't think we can move them all without a deprecation and major version change. 

Yep, I was suggesting we would be doing the config in this PR as part of the deprecation that all other extensions would have to go through. But I see the want to wait a bit to land the correct thing that would avoid adding more volume to the deprecation cycle. Thanks for #11105, hopefully we can get that in soon, and move this PR along :)